treatment-service/server/poller.go

```diff
@@ -0,0 +1,66 @@
+package server
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/caraml-dev/xp/treatment-service/config"
+	"github.com/caraml-dev/xp/treatment-service/models"
+)
+
+type Poller struct {
+	pollerConfig config.PollerConfig
+	localStorage *models.LocalStorage
+	stopChannel  chan struct{}
+	stopOnce     sync.Once
+	waitGroup    sync.WaitGroup
+}
+
+// NewPoller creates a new Poller instance with the given configuration and local storage.
+// pollerConfig: configuration for the poller
+// localStorage: local storage to be used by the poller
+func NewPoller(pollerConfig *config.PollerConfig, localStorage *models.LocalStorage) *Poller {
+	return &Poller{
+		pollerConfig: *pollerConfig,
+		localStorage: localStorage,
+		stopChannel:  make(chan struct{}),
+	}
+}
+
+func (p *Poller) Start() {
+	var startOnce sync.Once
+	startOnce.Do(func() {
+		ticker := time.NewTicker(p.pollerConfig.PollInterval)
+		p.waitGroup.Add(1)
+		go func() {
+			defer p.waitGroup.Done()
+			for {
+				select {
+				case <-ticker.C:
+					err := p.localStorage.Init()
```

Comment:
nit: `Init()` does not sound great to me, is it possible to rename it to maybe `Refresh`? Or just `Reinit` which internally call `init()`

Reply:
Yes, it makes sense since it does not reflect the intent. Changed it to `Refresh()`